src/PayzenGatewayFactory.php

```diff
@@ -42,7 +55,7 @@ protected function populateConfig(ArrayObject $config)
             'payum.action.refund'          => new Action\RefundAction(),
             'payum.action.status'          => new Action\StatusAction(),
             'payum.action.notify'          => new Action\NotifyAction(),
-            'payum.action.api.request'     => new Action\Api\ApiRequestAction(),
+            'payum.action.api.request'     => new Action\Api\ApiRequestAction($this->transactionId),
             'payum.action.api.response'    => new Action\Api\ApiResponseAction(),
```

Comment:
It might be possible to configure transaction id generator here.
Something like:
```php
            'payum.action.api.directory`      => sys_get_temp_dir(),
            'payum.action.api.id_generator`   => function(array $config) { 
                return new IdGeneratedByFile($config['payum.action.api.directory`]) 
            },
```

Reply:
I agree too, when you are using Payum, it's better to inject what you need on the Action or on the Api services. Here it's better to just inject it into `payum.action.api.request` as @ekyna said.